runtime/compiler/env/VMJ9.cpp

```diff
@@ -5886,6 +6034,12 @@ TR_J9VMBase::getClassFromNewArrayTypeNonNull(int32_t arrayType)
    return clazz;
    }
 
+bool
+TR_J9VMBase::getReportByteCodeInfoAtCatchBlock()
+   {
+   return _compInfoPT->getCompilation()->getOptions()->getReportByteCodeInfoAtCatchBlock();
```

Comment:
There are two usage points for `getOptions()->getReportByteCodeInfoAtCatchBlock()` which is a static and cannot be used at the server: `MetaData.cpp` and `J9CodeGenerator.cpp`
These usage points need to be replaced with this frontend query.
Alternatively, `static bool Options::_reportByteCodeInfoAtCatchBlock` needs not to be static.

BTW, how is this supposed to work:
```
TR::Options *
J9::Options::unpackOptions(char *clientOptions, size_t clientOptionsSize, TR::CompilationInfoPerThreadBase* compInfoPT, TR_J9VMBase *fe, TR_Memory *trMemory)
   {
   TR::Options *options = (TR::Options *)trMemory->allocateHeapMemory(clientOptionsSize);
   memcpy(options, clientOptions, clientOptionsSize);
   ...
   _reportByteCodeInfoAtCatchBlock = fe->getReportByteCodeInfoAtCatchBlock();
```
If _reportByteCodeInfoAtCatchBlock  is static, one client may clash with another.

Reply:
JITServer should not be checking the static field `_reportByteCodeInfoAtCatchBlock` on the server at all. So i guess unpackOptions shouldn't bother setting it, removing it from unpackOptions and replacing `getOptions()->getReportByteCodeInfoAtCatchBlock()` usage points on JITServer to the frontend query.